CONTRIBUTING.md

```diff
@@ -0,0 +1,316 @@
+Commit these changes before the major change, independently of that work.
+
+- If possible, decompose the remaining interrelated work into unrelated sets of changes.
+Next, define the first increment and get consensus on the development goal of the change.
+
+- Make each change in the set either stand alone (for example, to fix a bug) or part of a planned series of changes that work toward the development goal.
+Explaining these relationships to the community can be helpful.
+
+If you are interested in making a large change and feel unsure about its overall effect, please make sure to first discuss the change and reach a consensus through the developer forums.
+Then ask about the best way to go about making the change.
+
+### Commit Messages
+
+Although we don’t enforce a strict format for commit messages, we prefer that you follow the guidelines below, which are common among open source projects.
+Following these guidelines helps with the review process, searching commit logs, and email formatting.
+At a high level, the contents of the commit message should be to convey the rationale of the change, without delving into much detail.
+For example, “bits were not set right” leaves the reviewer wondering about which bits and why they weren’t “right”.
+In contrast, “Correctly compute ‘is dependent type’ bits in ‘Type’” conveys almost all there is to the change.
+
+Below are some guidelines about the format of the commit message itself:
+
+- Separate the commit message into a single-line title and a separate body that describes the change.
+- Make the title concise to be easily read within a commit log and to fit in the subject line of a commit email.
+- In changes that are restricted to a specific part of the code, include a [tag] at the start of the line in square brackets—for example, “[stdlib] …” or “[SILGen] …”.
+This tag helps email filters and searches for post-commit reviews.
+- When there is a body, separate it from the title by an empty line.
+- Make body concise, while including the complete reasoning.
+Unless required to understand the change, additional code examples or other details should be left to bug comments or the mailing list.
+- If the commit fixes an issue in the bug tracking system, include a link to the issue in the message.
+- For text formatting and spelling, follow the same rules as documentation and in-code comments—for example, the use of capitalization and periods.
+- If the commit is a bug fix on top of another recently committed change, or a revert or reapply of a patch, include the Git revision number of the prior related commit, for example “Revert abcdef because it caused bug#”.
+
+For minor violations of these guidelines, the community normally favors reminding the contributor of this policy over reverting.
+Minor corrections and omissions can be handled by sending a reply to the commits mailing list.
```

Comment:
I have no idea what this sentence is supposed to refer to – we don’t have a commits mailing list and if we do there is no significant traffic on it. Let’s delete it.

Reply:
dropping both sentences